File: server/routes/aqi.js
const express = require('express');
const router = express.Router();
const { getCitiesAQI, getCityAQI, getLocationAQI, searchStations, predictAQI } = require('../controllers/aqiController');

router.get('/cities', getCitiesAQI);          // All Indian cities real-time AQI
router.get('/city/:cityName', getCityAQI);    // Single city real-time AQI
router.get('/location', getLocationAQI);      // AQI by GPS lat/lng coordinates
router.get('/search', searchStations);        // Search AQI stations by keyword
router.get('/predict', predictAQI);           // 7-day AQI prediction

module.exports = router;

File: server/routes/aqi.py
from fastapi import APIRouter, Query
from controllers.aqi_controller import get_cities_aqi, get_city_aqi, get_location_aqi, search_stations, predict_aqi

router = APIRouter(prefix="/aqi")

@router.get("/cities")
async def cities():
    return await get_cities_aqi()

@router.get("/city/{city_name}")
async def city(city_name: str):
    return await get_city_aqi(city_name)

@router.get("/location")
async def location(latitude: float = Query(...), longitude: float = Query(...)):
    return await get_location_aqi(latitude, longitude)

@router.get("/search")
async def search(keyword: str = Query(...)):
    return await search_stations(keyword)

@router.get("/predict")
async def predict(city: str = Query(None)):
    return await predict_aqi(city)
